### PR TITLE
Handle QA searches async and independently.

### LINKED ISSUE #1356: Evaluate improving response speed on QA lookups
Review our code and look for performanc bottlenecks. (Up to 2 hour evaluating.)

Note in particular, the use of Promises.all(). 
Create additional followup tickets and link to #1346.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #1356

### CODE REVIEW

**Comment by justinlittman** [comment]
In local dev, an example of a InputLookupQA configured with multiple authorities is Monograph Work (BIBFRAME) > Subject of the Work.

You should expect:
* Results are displayed from authorities as they come in. (Rather than wait for all authorities to come in.)
* You should only see results from one search at a time. (So, as you modify your query, results from different searches shouldn't be mixed.)

File: src/actions/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Copyright 2019 Stanford University see LICENSE for license
-/* eslint max-params: ["warn", 4] */
+/* eslint max-params: ["warn", 5] */
```
Comment (jcoyne) [inline src/actions/index.js]: I don't understand why this should change if nothing else in the file changed.

File: src/components/editor/property/InputLookupQA.jsx [hunk 2]
```diff
@@ -23,16 +23,39 @@ const AsyncTypeahead = asyncContainer(Typeahead)
 //  based on values in propertyTemplate.valueConstraint.useValuesFrom
 //  and the lookupConfig for the URIs has component value of 'lookup'
 const InputLookupQA = (props) => {
-  const [authorities, setAuthorities] = useState([])
+  const triggerRenderState = useState('')
```
Comment (jcoyne) [inline src/components/editor/property/InputLookupQA.jsx]: can this be `const [_, setTriggerRender] = ...`

File: src/components/editor/property/InputLookupQA.jsx [hunk 3]
```diff
@@ -23,16 +23,39 @@ const AsyncTypeahead = asyncContainer(Typeahead)
...
+
+    // resultPromises is an array of Promise<authority>
+    const resultPromises = getSearchResults(query, props.propertyTemplate)
+    resultPromises.forEach((resultPromise) => {
+      resultPromise.then((authority) => {
+        // Only use these results if not cancelled.
+        if (!token.cancel) {
+          allAuthorities.current.push(authority)
+          // Changing state triggers re-render.
+          setTriggerRender(authority)
```
Comment (jcoyne) [inline src/components/editor/property/InputLookupQA.jsx]: Would there ever be a problem here if the `authority` for the last search is the same as the first authority for the next search? Particularly when a search only uses one authority?

File: src/utilities/qa.js [hunk 4]
```diff
@@ -66,7 +54,7 @@ export const createLookupPromises = (client, query, lookupConfigs) => lookupConf
    */
   const actionFunction = subauthority ? subAuthCall : authorityCall
 
-  return client
+  return Swagger({ spec: swaggerSpec }).then(client => client
```
Comment (jcoyne) [inline src/utilities/qa.js]: Does this cause the swagger to be reloaded every time this is called? Would that be slow?

**Pull request merged**